packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2729,6 +2729,42 @@ class BuildOwner {
   }
 }
 
+// Methods to handle hot reloads replacements of Stateless/Stateful Widget types.
+
+// If a StatelessWidget has been transformed into a stateful one, then accessing
+// `StatelessElement.widget` will throw a `CastError`. Instead, we access the
+// private member `Element._widget`. For a subclass of Element this field will
+// be defined and properly calling the element lifecycle methods will populate
+// it as normal. For implementations of Element, the assertion will never be
+// called.
+
+// Verify that an Element's widget is non-null.
+bool _debugSafeVerifyNonNullWidget(Element element) {
+  if (element is StatelessElement) {
```

Comment:
Why special-casing StatelessElement? Any reason not to always use _widget?

Reply:
Same below.

Reply:
If we can just use _widget, we also wouldn't need these helpers keeping the code slightly simpler?

Reply:
I created the methods to centralize documentation, but I could inline them all